File: src/main/java/com/github/xumeng/java/problems/easy/problem0342/Solution0342.java
package com.github.xumeng.java.problems.easy.problem0342;

public class Solution0342 {

    public boolean isPowerOfFour(int n) {
        if (n < 1) {
            return false;
        }
        double powerOfFour = 0.0d;
        for (int i = 0; powerOfFour <= Integer.MAX_VALUE; i++) {
            powerOfFour = Math.pow(4, i);
            if (n == powerOfFour) {
                return true;
            }
            if (n < powerOfFour) {
                return false;
            }
        }
        return false;
    }

}

File: src/main/kotlin/com/github/xumeng/kotlin/problems/easy/problem0342/Solution0342.kt
package com.github.xumeng.kotlin.problems.easy.problem0342

import kotlin.math.pow

class Solution0342 {

    fun isPowerOfFour(n: Int): Boolean {
        if (n < 1) {
            return false
        }
        var powerOfFour = 0.0
        var i = 0
        val base = 4.0
        while (powerOfFour < Int.MAX_VALUE) {
            powerOfFour = base.pow(i)
            if (n.toDouble() == powerOfFour) {
                return true
            }
            if (n < powerOfFour) {
                return false
            }
            i++
        }
        return false
    }

}